Repository validation fails with `NullPointerException`

### CrateDB version

6.4.0-SNAPSHOT-78a9e5e

### CrateDB setup information

Standard Docker single-node

### Problem description

When running [cratedb-toolkit tests](https://github.com/crate/cratedb-toolkit/blob/54bbfb1874fbf6d119c139c64b0eaac43c6a6294/tests/retention/test_cli.py#L271) locally against the latest CrateDB nightly build, I noticed an NPE during repository creation.

### Steps to Reproduce

Start a default Docker container and run:

```sql
CREATE REPOSITORY
                export_cold
            TYPE
                azure
            WITH (
                endpoint   = 'http://172.17.0.7:10000/devstoreaccount1',
                account    = 'devstoreaccount1',
                key        = 'Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==',
                container  = 'cratedb-cold-storage',
                base_path  = ''
            );
```

### Actual Result

Validating the repository location fails, which is expected:

```
[2026-06-18T13:59:04,020][ERROR][i.c.o.OpenDALBlobContainer] [Aiguille de Venosc] blobExists stat error on tests-HUiWQabLSEuj65THQbs1IQ/master.dat
org.apache.opendal.OpenDALException: Unexpected (persistent) at stat => send http request

Context:
   url: http://172.17.0.7:10000/devstoreaccount1/cratedb-cold-storage/tests-HUiWQabLSEuj65THQbs1IQ/master.dat
   called: http_util::Client::send
   service: azblob
   path: tests-HUiWQabLSEuj65THQbs1IQ/master.dat

Source:
   error sending request for url (http://172.17.0.7:10000/devstoreaccount1/cratedb-cold-storage/tests-HUiWQabLSEuj65THQbs1IQ/master.dat): client error (Connect): tcp connect error: No route to host (os error 113)

	at org.apache.opendal.Operator.stat(Native Method)
	at org.apache.opendal.Operator.stat(Operator.java:119)
	at io.crate.opendal.OpenDALBlobContainer.blobExists(OpenDALBlobContainer.java:65)
	at io.crate.opendal.OpenDALBlobContainer.writeBlob(OpenDALBlobContainer.java:86)
	at org.elasticsearch.common.blobstore.BlobContainer.writeBlobAtomic(BlobContainer.java:119)
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1186)
	at org.elasticsearch.repositories.RepositoriesService$4.doRun(RepositoriesService.java:376)
	at org.elasticsearch.common.util.concurrent.RejectableRunnable.run(RejectableRunnable.java:36)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
```

But the `CREATE REPOSITORY` statement returns in CrateDB with an NPE:

```

org.elasticsearch.repositories.RepositoryException: [] [export_cold] Unable to verify the repository, [export_cold] is not accessible on master node: NullPointerException 'null'
	at io.crate.execution.ddl.RepositoryService.convertRepositoryException(RepositoryService.java:128)
	at io.crate.execution.ddl.RepositoryService.lambda$execute$2(RepositoryService.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniComposeExceptionally.tryFire(CompletableFuture.java:1061)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:531)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2221)
	at io.crate.concurrent.FutureActionListener.onFailure(FutureActionListener.java:37)
	at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$1(TransportMasterNodeAction.java:173)
	at org.elasticsearch.action.ActionListener.lambda$withOnFailure$0(ActionListener.java:125)
	at org.elasticsearch.action.ActionListener$SimpleListener.onFailure(ActionListener.java:106)
	at org.elasticsearch.action.ActionListener.accept(ActionListener.java:55)
	at org.elasticsearch.action.ActionListener.accept(ActionListener.java:38)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:884)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:862)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:531)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2221)
	at io.crate.concurrent.FutureActionListener.onFailure(FutureActionListener.java:37)
	at org.elasticsearch.action.ActionRunnable.onFailure(ActionRunnable.java:90)
	at org.elasticsearch.common.util.concurrent.RejectableRunnable.run(RejectableRunnable.java:38)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: org.elasticsearch.repositories.RepositoryVerificationException: [export_cold] Unable to verify the repository, [export_cold] is not accessible on master node: NullPointerException 'null'
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1195)
	at org.elasticsearch.repositories.RepositoriesService$4.doRun(RepositoriesService.java:376)
	at org.elasticsearch.common.util.concurrent.RejectableRunnable.run(RejectableRunnable.java:36)
	... 3 more
Caused by: java.lang.NullPointerException
	at org.apache.opendal.OperatorOutputStream.writeBytes(Native Method)
	at org.apache.opendal.OperatorOutputStream.flush(OperatorOutputStream.java:73)
	at org.apache.opendal.OperatorOutputStream.write(OperatorOutputStream.java:61)
	at java.base/java.io.OutputStream.write(OutputStream.java:167)
	at java.base/java.io.InputStream.transferTo(InputStream.java:796)
	at io.crate.opendal.OpenDALBlobContainer.writeBlob(OpenDALBlobContainer.java:90)
	at org.elasticsearch.common.blobstore.BlobContainer.writeBlobAtomic(BlobContainer.java:119)
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1186)
	... 5 more
	Suppressed: org.apache.opendal.OpenDALException: Unexpected (persistent) at write => send http request

Context:
   url: http://172.17.0.7:10000/devstoreaccount1/cratedb-cold-storage/tests-HUiWQabLSEuj65THQbs1IQ/master.dat?comp=block&blockid=8GstyAhLQHyHYAt37TBl2w%3D%3D
   called: http_util::Client::send
   service: azblob
   path: tests-HUiWQabLSEuj65THQbs1IQ/master.dat
   size: 22
   written: 22

Source:
   error sending request for url (http://172.17.0.7:10000/devstoreaccount1/cratedb-cold-storage/tests-HUiWQabLSEuj65THQbs1IQ/master.dat?comp=block&blockid=8GstyAhLQHyHYAt37TBl2w%3D%3D): client error (Connect): tcp connect error: No route to host (os error 113)

		at org.apache.opendal.OperatorOutputStream.writeBytes(Native Method)
		at org.apache.opendal.OperatorOutputStream.flush(OperatorOutputStream.java:73)
		at org.apache.opendal.OperatorOutputStream.close(OperatorOutputStream.java:80)
		at io.crate.opendal.OpenDALBlobContainer.writeBlob(OpenDALBlobContainer.java:89)
		... 7 more
```

### Expected Result

A proper error message without an NPE

--- a/docs/appendices/release-notes/6.3.5.rst
+++ b/docs/appendices/release-notes/6.3.5.rst
@@ -46,4 +46,6 @@ series.
 Fixes
 =====
 
-None
+- Fixed an issue that would throw a misleading ``NullPointerException`` when
+  ``CREATE REPOSITORY`` failed during repository verification, by returning a
+  user friendly error message.

--- a/libs/opendal/pom.xml
+++ b/libs/opendal/pom.xml
@@ -48,6 +48,12 @@
             <scope>test</scope>
         </dependency>
         <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>${versions.mockito}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter</artifactId>
             <scope>test</scope>

--- a/libs/opendal/src/main/java/io/crate/opendal/OpenDALBlobContainer.java
+++ b/libs/opendal/src/main/java/io/crate/opendal/OpenDALBlobContainer.java
@@ -29,8 +29,6 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 import org.apache.opendal.OpenDALException;
 import org.apache.opendal.OpenDALException.Code;
 import org.apache.opendal.Operator;
@@ -38,8 +36,6 @@ import org.elasticsearch.common.blobstore.BlobContainer;
 import org.elasticsearch.common.blobstore.BlobPath;
 
 public class OpenDALBlobContainer implements BlobContainer {
-
-    private static final Logger LOGGER = LogManager.getLogger(OpenDALBlobContainer.class);
 
     private final BlobPath path;
     private final String pathAsString;
@@ -65,10 +61,10 @@ public class OpenDALBlobContainer implements BlobContainer {
             operator.stat(path);
             return true;
         } catch (OpenDALException ex) {
-            if (ex.getCode() != Code.NotFound) {
-                LOGGER.error("blobExists stat error on {}", path, ex);
+            if (ex.getCode() == Code.NotFound) {
+                return false;
             }
-            return false;
+            throw ex;
         }
     }
 

--- a/libs/opendal/src/test/java/io/crate/opendal/OpenDALBlobContainerTest.java
+++ b/libs/opendal/src/test/java/io/crate/opendal/OpenDALBlobContainerTest.java
@@ -22,6 +22,9 @@
 package io.crate.opendal;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.nio.file.Path;
 import java.util.List;
@@ -29,6 +32,8 @@ import java.util.Map;
 
 import org.apache.opendal.AsyncExecutor;
 import org.apache.opendal.AsyncOperator;
+import org.apache.opendal.OpenDALException;
+import org.apache.opendal.OpenDALException.Code;
 import org.apache.opendal.Operator;
 import org.apache.opendal.ServiceConfig;
 import org.elasticsearch.common.blobstore.BlobPath;
@@ -49,6 +54,26 @@ public class OpenDALBlobContainerTest extends ESTestCase {
     @After
     public void teardown() {
         executor.close();
+    }
+
+    @Test
+    public void test_blobExists_returns_false_on_not_found() throws Exception {
+        Operator operator = mock(Operator.class);
+        when(operator.stat("file1.txt")).thenThrow(new OpenDALException(Code.NotFound, "not found"));
+        OpenDALBlobContainer container = new OpenDALBlobContainer(BlobPath.cleanPath(), operator, 1024);
+
+        assertThat(container.blobExists("file1.txt")).isFalse();
+    }
+
+    @Test
+    public void test_blobExists_propagates_non_not_found_stat_failures() {
+        var failure = new OpenDALException(Code.Unexpected, "network failure");
+        Operator operator = mock(Operator.class);
+        when(operator.stat("file1.txt")).thenThrow(failure);
+        OpenDALBlobContainer container = new OpenDALBlobContainer(BlobPath.cleanPath(), operator, 1024);
+
+        assertThatThrownBy(() -> container.blobExists("file1.txt"))
+            .isSameAs(failure);
     }
 
     @Test

